File: works/1929.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Collections;



public class Main
{
    final static int MAX = 1000010;
    static int[] p = new int[MAX];

    static void FindPrime()
    {
        for(int i = 2; i < MAX; i++) p[i] = i;

        for(int i = 2; i * i < MAX; i++)
        {
            if(p[i] == 0) continue;
            for(int j = i*i; j < MAX; j += i) p[j] = 0;
        }
    }

    public static void main(String[] args) throws IOException
    {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();

        int n = sc.nextInt();
        int m = sc.nextInt();

        FindPrime();
        for(int i = n; i <= m; i++)
        {
            if(p[i] == 0) continue;
            sb.append(p[i]).append('\n');
        }
        System.out.print(sb);
    }
}

File: works/1929.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*
import kotlin.math.sqrt

val MAX = 1000010
var p = Array(MAX) {0}

fun FindPrime()
{
    for(i in 2 until MAX) p[i] = i

    var d = sqrt(1000010.0)
    var d_i = d.toInt()
    for(i in 2 until d_i)
    {
        if(p[i] == 0) continue
        for(j in i*i until MAX step i) p[j] = 0
    }
}

fun main()
{
    val sc = Scanner(System.`in`)
    val sb = StringBuilder()

    val n = sc.nextInt()
    val m = sc.nextInt()

    FindPrime()
    for(i in n .. m)
    {
        if(p[i] == 0) continue
        sb.append(p[i]).append('\n')
    }
    print(sb)
}